### PR TITLE
Changes to Grid Layout

### PR DESCRIPTION
**Pull request by subbramanil**
# Provides Layout Options

- Provides option to switch between List/Grid view
  - Grid View lists the apps in 2x2 grid view
- Shows the app icon along with the app name

Launch Setup Screen

Choose Layout

Apps In List

Apps In Grid

### CODE REVIEW

**Comment by subbramanil** [comment]
You may hold on to this PR for a while. I don't want to introduce breaking changes. I am planning to add a choice to pick between list/grid. It may take a while to finish.

**Comment by sduduzog** [comment]
This looks really neat, when I get the chance I'll also play around with it

**Comment by sduduzog** [comment]
Gives me something to do over the holidays. Like I'm interested in seeing if I can centre the odd (when last is 1st, 3rd or 5th) app on the grid.

**Comment by subbramanil** [comment]
Updated with options to switch between List and Grid view.

**Comment by subbramanil** [comment]
Check out the readme in https://github.com/subbramanil/slim-launcher after merging this PR. 

I can submit another PR for the readme update. Don't want to confuse this PR for now.

File: app/src/main/res/layout/setup_fragment.xml [hunk 1]
```diff
@@ -120,11 +120,25 @@
...
+        android:id="@+id/opt3Text"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginLeft="8dp"
+        android:layout_marginEnd="32dp"
+        android:layout_marginRight="32dp"
         android:text="@string/setup_text_choose_your_preferred_clock"
         android:textAppearance="@style/TextAppearance.AppCompat.Medium"
-        app:layout_constraintEnd_toStartOf="@+id/ivTick2"
```
Comment (subbramanil) [inline app/src/main/res/layout/setup_fragment.xml]: Note that the '+' is missing after the changes. 

When you declare an id for the first time you use, '+'. When you refer an already declared id, you don't need to add '+'.

**Comment by subbramanil** [comment]
Some of the icons are square icons put inside of a circle and some are circular icons. So it might look different. You can try updating the layout style for the image view. 

The icons are fetched not created by the app. So you may not be able to customize the icons. But you can update the image layout to look similar.

**Comment by sduduzog** [comment]
The codebase has changed so much since I've come back here. I was hesitant in merging this PR because I felt it didn't move to the direction I had envisioned when creating the app. _in short, an icon-free minimalist home screen_ event the ones at the bottom still give me some amount of stress because I'm not sure how users of the app would feel about it.

**Comment by subbramanil** [comment]
Absolutely fine. I just wanted to play around your code base and get a Hacktober T-Shirt. 

Please feel free to shoot me an email if you need any help related to android development. 

Keep up the good work!!

**Pull request closed without merging**